src/test/java/com/google/javascript/clutz/partial/destruct_ctor.d.ts

```diff
@@ -0,0 +1,11 @@
+declare namespace ಠ_ಠ.clutz.module$exports$derived$interface$union$undefined {
+  interface Derived extends ಠ_ಠ.clutz.module$exports$some$base.Base extends ಠ_ಠ.clutz.module$exports$some$base.Base {
+    someField ? : string ;
+  }
+  //!! Only a is generated correctly, b and c should be unioned with Derived.
+  function fn (a : ಠ_ಠ.clutz.module$exports$derived$interface$union$undefined.Derived , b : null , c ? : undefined ) : void ;
+}
+declare module 'goog:derived.interface.union.undefined' {
+  import undefined = ಠ_ಠ.clutz.module$exports$derived$interface$union$undefined;
```

Comment:
Can you choose a different name that doesn't conflict with built-in global identifier 'undefined'?

Reply:
done, changed also file name.